[문자열 덧셈 계산기] 김건학 미션 제출합니다.

Gonagi opened the pull request.

Gonagi committed:
docs: 기능목록 작성

Gonagi committed:
docs(ReadMe): 패키지 구조 추가

Gonagi committed:
docs(ReadMe): 문자열 유효성 검증 기능 수정

모든 조건이 동일하게 IllegalArgumentException을 throw하기 때문에 문자열 유효성 검증 로직을 하나의
기능으로 통합했습니다.

Gonagi committed:
feat(input): 문자열 입력 및 유효성 검증 기능 구현

getInput과 validateInput 기능 구현:
- getInput: 사용자로부터 문자열을 입력받고 validateInput을 호출하여 유효성 검증
- validateInput: 입력값이 null이거나 비었을 때, "//"로 시작하지 않을 때, "\n"이 없을 때 예외 발생

Gonagi committed:
style(input): 파일명 변경

stringInput.java의 파일명을 StringInput.java로 변경

Gonagi committed:
test(input): 입력한 문자열의 유효성 검증 테스트 코드 추가

validateInput 테스트:
- 유효한 입력에 대해 정상적으로 문자열이 반환되는지 검증
- 유효하지 않은 입력에 대해 IllegalArgumentException을 던지는지 검증
  - null 입력인 경우
  - 빈 문자열인 경우
  - 문자열이 '//'로 시작하지 않는 경우
  - 문자열에 '\n'이 없는 경우

Gonagi committed:
refactor(input): 파일명 변경

파스칼 표기법 적용
stringInputTest에서 StringInputTest로 변경

Gonagi committed:
docs(README): 문자열 자르는 기능 설명 추가 및 완료 상태 업데이트

Gonagi committed:
test(split): 문자열 자르는 기능의 테스트 코드 추가

개행문자를 기준으로 앞 문자열과 뒷 문자열이 정상적으로 잘렸는지 검증
첫번째로 나오는 개행문자로 문자열이 잘리는지 확인
개행문자의 뒷 부분이 빈 문자열인 경우 확인

Gonagi committed:
feat(split): 문자열 자르는 기능 구현

getFrontString과 getBackString 기능 구현:
- getFrontString: 개행문자를 기준으로 문자열을 자르고 앞부분의 문자열을 반환
- getBackString: 개행문자를 기준으로 문자열을 자르고 뒷부분의 문자열을 반환

Gonagi committed:
refactor(split): getFrontString 반환값 수정

getFrotString을 통해 얻은 문자열은 '구분자'로만 구성된다.
앞 문자열의 시작 부분인 '//'를 제거하여,  '//'와 '\n'이 제거된 순수한 문자열을 반환받는다.

Gonagi committed:
docs(README): 구분자를 가진 문자열 검증 기능 설명 수정

구분자를 가진 문자열 검증 기능 설명을 더 명확하게 수정하였습니다.
문자열 내 숫자가 포함된 경우 IllegalArgumentException을 던지는 로직을 분리했습니다.

Gonagi committed:
docs(README): 커스텀 구분자 기능 구현 및 메서드명 수정

커스텀 구분자 기능을 구현하여 체크박스를 체크합니다.
구분자 배열을 반환하는 기능의 메서드명을 registerSeperator에서 getSeperatorArray로 수정했습니다.

Gonagi committed:
test(split): 커스텀 구분자 정하는 기능의 테스트 코드 작성

개행문자를 기준으로 나눈 앞 문자열에 숫자가 있는지 검증
문자로만 이루어진 문자열의 각 문자를 추출하여 char 배열로 반환하는 테스트 코드 작성

Gonagi committed:
feat(split): 커스텀 구분자 정하는 기능 구현

구분자 문자열은 문자로만 구성되어야 하므로, 숫자가 포함되어 있는지 확인하는 기능을 구현했습니다. (validateFrontString)
구분자 문자열의 각 문자를 개별 구분자로 등록한 char 배열 반환하는 기능 구현했습니다. (getSeperatorArray)

Gonagi committed:
docs(README): 각 기능의 패키지 변경에 따른 기능 목록 수정

코드 구조를 보다 명확하게 하여 유사한 기능들을 그룹화하고, 코드의 가독성과 유지보수성을 향상시키기 위해 input 패키지에 있는
validateInput과 split 패키지에 있던 validateFrontString을 validate 패키지로 이동했습니다.

Gonagi committed:
refactor(calculator): 기능의 패키지 변경에 따른 리팩토링

코드 구조를 명확히 하기 위해 기존 코드 구조를 리팩토링했습니다.
input 패키지에 있던 validateInput과 split 패키지에 있던 validateFrontString을 validate
패키지로 이동시켰습니다.
StringInputTest.java를 삭제하고 StringValidateTest를 작성했습니다.

Gonagi committed:
docs(README): 커스텀 구분자 배열 생성 기능 삭제 및 패키지 구조 변경

문자열에서 구분자 추출하는 단계:
- 불필요한 배열 생성을 방지하기 위해 커스텀 구분자 배열 생성 기능을 삭제했습니다.
- 문자열에서 배열을 생성하지 않고, `StringSplit.getFrontString`에서 반환한 String을 그대로
사용하도록 변경했습니다

계산하는 문자열 검증하는 단계:
- IllegalArgumentException을 던지는 기능들을 하나로 묶었습니다.

패키지 구조를 수정:
- 파스칼 표기법을 적용하여 클래스 및 메소드 명명 규칙을 일관되게 유지했습니다.
- 기능 목록에 맞춰 패키지를 정리했습니다.

Gonagi committed:
refactor(split): 커스텀 구분자 배열 생성 기능 삭제

계산해야 하는 문자열에서 각 문자가 구분자인지 확인하는 과정에서 char 배열 대신 String의 contains 메소드를
사용하도록 변경했습니다.
이로써 `getFrontString`을 통해 얻은 원본 String을 그대로 활용하여, 불필요한 배열 생성을 줄이고 코드의
간결성을 높였습니다.

BREAKING CHANGE: 커스텀 구분자 배열 생성 기능이 삭제되어, 더 이상 char []  배열을 사용하지 않음
구분자 처리는 getFrontString에서 반환된 String을 사용하도록 변경함
구분자 문자열 관련해서 char 배열을 사용하는 코드가 독장하지 않을 수 있으며, String.contains로 대체해야 함

Gonagi committed:
docs(README): 계산해야 하는 문자열의 유효성을 검증하는 기능 업데이트

'앞 문자열', '숫자를 추출해야 하는 문자열'이란 표현을 '계산해야 하는 문자열'로 수정
기능 구현 완료

Gonagi committed:
test(validate): 계산해야 하는 문자열을 검증하는 기능의 테스트 코드 작성

계산해야 하는 문자열이 숫자, 구분자로만 구성되어 있는지 검증한다.

Gonagi committed:
feat(validate): 계산해야 하는 문자열을 검증하는 기능 구현 완료

계산할 문자열과 구분자 문자열을 입력받아, 해당 문자열이 구분자와 숫자로만 구성되었는지 검증한다.

Gonagi committed:
test(split): 오타 수정

'문자열가'를 '문자열이'로 수정

Gonagi committed:
fix(validate): 계산해야하는 문자열이 빈 문자열일 때 발생하는 out of bounds 오류 해결

계산할 문자열이 빈 경우 split 과정에서 발생하는 오류를 수정했습니다.
계산할 문자열이 빈 문자열인 경우, 바로 true를 반환하도록 로직 변경

Gonagi committed:
test(split): assertEquals 매개변수 위치 수정

assertEquals에서 expected와 result의 매개변수 위치가 반대로 설정되어 있던 문제를 확인하여, 올바른 순서인
expected, actual로 수정했습니다.

Gonagi committed:
style(validate): 오타 수정

'seperator'를 'separator'로 수정했습니다.

Gonagi committed:
test(validate): 테스트 코드 오타 수정

파스칼 표기법을 적용해 'onlyseparatorInBackString'를 'onlySeparatorInBackString'로
수정했습니다.

Gonagi committed:
docs(README): 계산해야 하는 문자열의 숫자만 추출하여 배열만드는 기능 수정 및 업데이트

문자열을 분리하여 숫자로 변환한 배열을 생성하는 기능을 두 개의 기능으로 분리했습니다.
각각의 기능은 문자열을 분리하여 배열로 만드는 기능과 문자열을 숫자로 변환하는 기능입니다.
해당 기능 구현을 완료했습니다.

Gonagi committed:
test(split): 계산할 문자열에서 숫자를 추출해 배열로 변환하는 기능에 대한 테스트 코드를 작성

splitStringBySeparator:
- 구분자로 문자열을 분리하고, 분리된 숫자를 배열로 반환하는 기능에 대한 테스트 작성
- 계산할 문자열이 빈 경우, 배열 [0]을 반환하는 테스트 코드 추가
- 계산할 문자열이 구분자로만 이루어진 경우, 0으로 채워진 배열을 반환하는 테스트 작성

stringToNumber:
- 문자열을 숫자로 변환하는 기능의 테스트 코드 작성
- 빈 문자열을 숫자 0으로 변환하는 테스트 추가

Gonagi committed:
feat(split): 계산할 문자열에서 숫자를 추출해 배열로 변환하는 기능 구현

splitStringBySeparator:
- 구분자로 문자열을 나누고, 각 구간을 숫자로 변환한다.
- 구분자가 연속으로 등장하거나 빈 구간은 0을 반환한다.
- 구분자 사이에 숫자가 있으면 해당 숫자를 반환한다.

stringToNumber:
- 문자열을 숫자로 변환한다.
- 빈 문자열인 경우 0을 반환한다.

Gonagi committed:
docs(README): 숫자를 더하고 출력하는 기능 구현

Gonagi committed:
test(calculate): 숫자를 더하고 출력하는 기능에 대한 테스트 코드 작성

getResult:
- 빈 int 배열인 경우 0을 반환하는 테스트 코드 작성
- int 배열의 모든 요소를 순회하며 합계를 반환하는 기능에 대한 테스트 코드 작성

Gonagi committed:
feat(calculate): 숫자를 더하고 출력하는 기능에 구현

getResult:
- 빈 int 배열인 경우 0을 반환한다.
- 그 외의 경우 int 배열의 모든 요소를 순회하며 합계를 반환한다.

printResult:
- "결과 : result"를 출력한다.

Gonagi committed:
docs(README): 기능 목록 업데이트

기본 구분자와 커스텀 구분자가 따로 있다는 것을 인지하지 못해 수정했습니다.
- 입력받은 문자열 검증 기능 분리 (isInputNull, isInputStartWithSlash)
- "//"로 시작하는 경우와 그렇지 않은 경우에 따른 문자열 검증 기능 분리 (isNewLineInInput, isInputStartWithNumOrBasicSeparator)
- 구분자를 등록하는 기능 분리 기본(getDefaultSeparator, getDefaultCalFormula), 커스텀(getCustomSeparator, getCustomCalFormula)
- 구분자로 구성된 문자열과 계산해야 하는 문자열을 검증하는 기능명 변경 (validateSeparator, validateCalFormula)
- 계산해야 하는 문자열에서 숫자만 추출하여 배열을 만드는 기능명 변경 (splitCalFormulaBySeparator)
- 문자열을 숫자로 변환하는 기능명 변경 (stringToNum)
- 배열의 숫자를 더하는 기능명 변경 (getNumSum)
- 결과 출력 기능명 변경 (printResult)

Gonagi committed:
fix(README, input, validate): 문자열 입력 및 검증 오류 수정

입력한 문자열이 null인 경우 오류를 던집니다.
빈 문자열 입력 시 오류를 던지지 않도록 수정했습니다.
문자열이 "//"로 시작하면 커스텀 구분자 기능을 사용하고, 그 외의 문자나 숫자로 시작하면 기본 구분자로 구분하도록 수정했습니다.
문자열이 반드시 "//"로 시작할 필요 없으며, "\n"을 포함하지 않아도 되도록 수정했습니다.

Gonagi committed:
fix(README, split): 구분자 등록 기능명 수정 및 기능 추가

입력받은 문자열의 형태에 따라 구분자를 반환하는 기능을 수정했습니다.
문자열이 "//"로 시작하는 경우, getCustomSeparator로 커스텀 구분자와 기본 구분자를 등록하고,
getCustomCalFormula로 계산할 문자열을 추출합니다.
문자열이 숫자로 시작하는 경우, getDefaultSeparator로 기본 구분자를 등록하고,
getDefaultCalFormula로 계산할 문자열을 추출합니다.

Gonagi committed:
refactor(README, validate): 구분자 및 계산식 검증 기능의 반환형 수정 및 메소드명 변경

구분자 및 계산식 검증 기능의 반환형을 boolean에서 void로 변경하여 if문으로 처리하지 않아도 되도록 수정했습니다.
메소드명을 더 직관적으로 변경했습니다.
구분자 검증 기능의 메소드명을 validateFrontString에서 validateSeparator로 수정했습니다.
계산식 검증 기능의 메소드명을 validateBackString에서 validateCalFormula로 수정했습니다.

Gonagi committed:
refactor(README, split): 정수형 배열을 만드는 메소드명 및 매개변수 변경

메소드명과 매개변수를 더 직관적으로 변경했습니다.
배열 생성 메소드명을 splitCalFormulaBySeparator에서 splitCalFormulaBySeparator로 수정했습니다.
배열 생성 메소드 내 계산식을 넘기는 매개변수를 string에서 calFormula로 수정했습니다.
문자열을 숫자로 만드는 메소드 내 문자열을 넘기는 매개변수를 string에서 strNum으로 수정했습니다.

Gonagi committed:
refactor(README, calculator): 숫자 배열 합산 메소드명 변경

메소드명을 getResult에서 getNumSum으로 변경하여 기능을 더 직관적으로 표현했습니다.

Gonagi committed:
refactor(application): 클래스 및 메소드 리팩토링에 맞춰 로직 수정

입력에 따른 커스텀 구분자 및 기본 구분자 처리를 명확하게 구분했습니다.
구분자와 계산식 검증 로직을 분리하고 각 기능을 더 직관적으로 처리할 수 있도록 리팩토링했습니다.

Gonagi committed:
fix(README, split, validate): 문자열을 분리하는 문자열 처리 개선

"\\n"은 이스케이프 문자 "\"와 "n"으로 처리되어 오류가 발생했습니다.
validate와 split에서 두가지 형식 모두 처리할 수 있도록 수정했습니다.
문자열을 분리하는 문자열의 후보로 개행문자와 "\"+"n"를 추가했습니다.
문자열을 분리하는 정규표현식으로 "\\\\n|\\n"를 사용했습니다.

Gonagi committed:
refactor(application, README, split): 구분자와 계산식 처리 방식 개선 및 검증 일관성 향상

변경되지 않는 값인 regex와 defaultSeparator를 static final로 상수 처리했습니다.
기본 구분자와 커스텀 구분자를 따로 처리하지 않고, 하나의 과정으로 통합하여 처리하도록 리팩토링했습니다.
유효한 구분자와 계산식을 검증하는 로직을 splitCalFormulaBySeparator 메서드 내부로 이동시켜, 일관성을
유지했습니다.

BREAKING CHANGE:
getCustomSeparator 및 getDefaultSeparator 메서드가 getSeparator로 변경되었습니다.
getCustomCalFormula 및 getDefaultCalFormula 메서드가 getCalFormula로 변경되었습니다.

StoneCAU commented on README.md:
리드미 파일을 프로그램 플로우에 따라 꼼꼼하게 작성하신게 인상깊습니다!

Gonagi committed:
docs(README): 리팩토링에 따른 README 수정

각 패키지에 따라 기능을 나눴습니다.

StoneCAU commented on src/main/java/calculator/split/StringSplit.java:
```diff
@@ -0,0 +1,47 @@
+import java.util.Objects;
+
+public class StringSplit {
+    public static final String REGEX = "(\\\\n|\\n)";
+    public static final String DEFAULT_SEPARATOR = ",:";
```
해당 클래스에서만 쓰이는 것 같은데 private로 하시는건 어떨까요?

Gonagi replied:
좋은 지적 감사합니다.

StoneCAU commented on src/main/java/calculator/validate/StringValidate.java:
```diff
@@ -0,0 +1,44 @@
+        return input.startsWith("//");
+    }
+
+    public static void validateNewLineInput(String input) {
+        if (input.contains("\n") || input.contains("\\n")) {
+            return;
+        }
```
입력에서 "\n"과 "\\n" 모두 동일한 기능을 하는건지 궁금합니다!

Gonagi replied:
문자열에 \가 오면 뒤에 오는 문자에 따라 그 문자가 다르게 변환됩니다.
- 입력이 `//?\n1,2?3`일 때, input은 `//? + 개행 문자 + 1,2?3`으로 변환됩니다.
- 입력이 `//?\\n1,2?3`일 때, input은 `//? + 이스케이프 문자 + n + 1,2?3`으로 변환됩니다.

첫 번째 예시처럼 일반적인 경우는 `input.contains("\n")`로 개행 문자를 검증할 수 있지만, 두 번째 예시처럼 `\\n`이 있을 경우는 잡지 못합니다. 
따라서, 추가 조건으로 input.contains("\\n")을 추가하여 두 번째 예시의 `이스케이프 문자 + n`을 검출했습니다.

StoneCAU commented on src/main/java/calculator/split/StringSplit.java:
```diff
@@ -0,0 +1,47 @@
+        return string;
+    }
+
+    public static int[] splitCalFormulaBySeparator(String calFormula, String separator) {
+        validateSeperator(separator);
+        validateCalFormula(calFormula, separator);
+        return Arrays.stream(calFormula.split(String.format("[%s]", separator), -1))
+                .mapToInt(StringSplit::stringToNumber)
+                .toArray();
+    }
```
검증과정이 깔끔하게 잘 이루어진 것 같습니다!!

StoneCAU reviewed (commented):
우선 제 코드에 리뷰 남겨주셔서 감사드리고 1주 차 고생하셨습니다😀 리드미 부분과 검증과정이 인상깊었습니다! 개인적으로 public과 private 부분만 잘 고려해서 작성해주시면 더 좋은 코드가 될 것 같습니다.

Gonagi committed:
refactor(split): 기본 구분자의 접근 지정자를 private로 수정

StringSplit.java에서만 사용되는 상수이므로, 접근 지정자를 private로 변경했습니다.

Gonagi committed:
refactor(split): 모든 메소드의 매개변수에 final 키워드를 추가하여 불변성 보장

데이터의 정합성을 유지하였다.

neoskyclad reviewed (commented):
문제를 분석하여 README에 자세하게 작성하신 것이 인상 깊습니다😊

ykmxxi commented on src/main/java/calculator/Application.java:
```diff
@@ -1,7 +1,24 @@
+
 public class Application {
     public static void main(String[] args) {
         // TODO: 프로그램 구현
```
TODO의 목적인 구현을 끝냈으니 지워도 되는 주석인것 같습니다ㅎ

Gonagi replied:
넵 🙈

ykmxxi commented on src/main/java/calculator/Application.java:
```diff
@@ -1,7 +1,24 @@
 public class Application {
     public static void main(String[] args) {
         // TODO: 프로그램 구현
+        String input = getInput();
+
+        String separator = getSeparator(input);
+        String calFormula = getCalFormula(input);
+
+        int[] numArr = splitCalFormulaBySeparator(calFormula, separator);
+        int result = getNumSum(numArr);
+
+        printResult(result);
```
문맥에 따라 공백 라인으로 분리된 코드가 좋은것 같습니다

궁금한 점은 모든 클래스의 메서드를 정적 메서드로 만드셨는데 특별한 이유가 있으셨나요?

Gonagi replied:
'문자열 계산기'가 단순한 예제라 생각해 객체의 필요성을 못느껴서 모든 메서드를 정적 메서드로 작성했습니다. 
그러나 코드 리뷰를 통해 다른 분들의 코드를 살펴보니, '문자열 계산기'는 결코 단순하지 않으며, 많은 분들이 확장성을 고려한 객체 지향적인 방식으로 구현한 것을 확인했습니다. 
이에 저도 객체 지향적으로 리팩토링할 계획입니다.

ykmxxi commented on src/main/java/calculator/split/StringSplit.java:
```diff
@@ -0,0 +1,47 @@
+    }
+
+    public static int stringToNumber(String strNum) {
+        if (Objects.equals(strNum, "")) {
```
혹시 발생할지 모를 null을 처리하기 위해 사용하신건가요?

Gonagi replied:
저는 구분자가 연속으로 나오는 경우, 구분자가 처음 또는 마지막에 나오는 경우를 정상적인 입력으로 처리했습니다. 이러한 경우는 ""으로 변환되며, ""은 0으로 변환해야 하므로 위와 같이코드를 짰습니다.
- `1::2:3`를 입력하면 ["1",  "",  "2", "3"]배열을 생성후, `stringToNumber`을 이용해  [1, 0, 2, 3]을 생성합니다.
- `:1:2:3:`을 입력하면  ["", "1", "2", "3", ""]배열을 생성,  [0, 1, 2, 3, 0]배열을 생성합니다. 
- 추가로 `""`을 입력했을 때도 [""] 배열 생성 후, [0]을 생성합니다.

ykmxxi replied:
딱 처음 코드를 봤을 때는 null 처리에 대한 고민이 있어서 Objects.equals()를 사용하셨나? 생각이 들어 여쭤봤었습니다.

`"".equals(strNum)`를 사용하셔도 NPE를 방지하면서 동등성 검증을 할 수 있어 한 번 추천드립니다

ykmxxi commented on src/main/java/calculator/validate/StringValidate.java:
```diff
@@ -0,0 +1,44 @@
+        throw new IllegalArgumentException("문자열에 '\n' 또는 '\\n'이 있어야 합니다.");
+    }
+
+    public static void validateSeperator(String seperator) {
```
이 코드도 위에서 작성하신 `StringSplit.splitCalFormulaBySeparator()`메서드 처럼 고수준의 함수로 처리할 수 있을 것 같습니다

Gonagi replied:
수정하겠습니다!

ykmxxi commented on src/test/java/calculator/calculate/CalculatorTest.java:
```diff
@@ -0,0 +1,26 @@
+
+        Assertions.assertEquals(0, getNumSum(arr));
+    }
+}
```
EOF 추가가 되지 않은 파일이 존재합니다. 사용 하시는 IDE에서 저장 시 자동으로 EOF를 추가해주는 기능을 활용해보시면 좋다고 생각합니다

ykmxxi commented on src/main/java/calculator/input/StringInput.java:
```diff
@@ -0,0 +1,15 @@
+public class StringInput {
+    public static String getInput() {
+        System.out.println("덧셈할 문자열을 입력해 주세요.");
+        String input = Console.readLine();
```
자원 해제를 한번 고민해보시면 좋을것 같습니다

Gonagi replied:
확인했습니다.

ykmxxi commented on src/test/java/calculator/split/StringSplitTest.java:
```diff
@@ -0,0 +1,145 @@
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+class StringSplitTest {
```
테스트 케이스를 꼼꼼히 설정해 작성하셔서 메서드가 어떻게 동작하는지 쉽게 알 수 있는 좋은 테스트인것 같습니다.

여기서 중복되는 형식의 검증을 `@ParameterizedTest`를 사용해 처리하신다면 더 읽기 좋은 테스트가 될것 같습니다

Gonagi replied:
넵 덕분에 테스트 코드 내 중복되는 검증 로직을 효율적으로 처리하도록 개선할 수 있었습니다.

ykmxxi reviewed (commented):
리뷰 남깁니다. 클래스와 메서드 이름이 직관적이라 읽기 좋은 코드였습니다

Gonagi committed:
refactor(input): 문자열 입력 후 쓰지않는 자원을 해제하도록 수정

Java 7 이상부터 지원하는 try-with-resources를 사용하여, finally 블록에서 직접 자원을 해제하지 않고도
자동으로 자원이 해제되도록 하였습니다.

Gonagi committed:
refactor(validate): 분리자 문자열 검증 로직을 Stream API를 사용하여 리팩토링

for-each 루프 대신 chars()와 anyMatch()를 사용하여 숫자 포함 여부를 검사하도록 수정했습니다.

Gonagi committed:
style(calculate, split, validate): 파일 끝에 EOF 추가

사용하는 IDE에서 저장 시 자동으로 EOF를 추가해주는 기능을 활용하라는 의견을 반영하여 설정을 변경했습니다.
POSIX를 준수합니다.

Gonagi committed:
test(calculate, split, validate): @ParameterizedTest를 사용하여 중복 검증 로직 개선

테스트 코드 내 중복되는 검증 로직을 @ParameterizedTest, @MethodSource, @ValueSource를
사용해 보다 효율적으로 관리하도록 개선했습니다.
- 동일한 패턴의 입력과 검증 로직을 @ParameterizedTest로 대체하여 코드 가독성 향상
- MethodSource와 ValueSource를 통해 여러 입력 값과 예상 결과를 전달하여 테스트 케이스를 간소화

Gonagi committed:
style(application): 주석 삭제

필요없는 주석 삭제

Gonagi reviewed (commented):
많은 도움 주셔서 감사합니다.
